### PR TITLE
feat: add GET_PROTOCOL_PARAMETERS

### PR DESCRIPTION
**Pull request by iccicci**
This should solve #254

### CODE REVIEW

File: test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        a0: 0.3,
+        rho: 0.003,
+        tau: 0.2,
+        decentralisation_param: 0,
+        extra_entropy: null,
+        protocol_major_ver: 9,
+        protocol_minor_ver: 0,
+        min_utxo: '4310',
+        min_pool_cost: '170000000',
+        nonce: '538e6395f3e5c8b415ac054e56ce4bfed41ce39c0fd2890c86d646a4a387b9b9',
```
Comment (slowbackspace) [inline test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts:27]:
```suggestion
        nonce: expect.any(String),
```

this changes every epoch, same as epoch number

File: test/e2e/tests/get-protocolp-parameters.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { describe, expect, test } from 'vitest';
```
Comment (slowbackspace) [inline test/e2e/tests/get-protocolp-parameters.e2e.test.ts:1]: typo in filename? get-protocol**p**-parameters.e2e.test.ts

File: test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        key_deposit: '2000000',
+        pool_deposit: '500000000',
+        e_max: 18,
+        n_opt: 500,
+        a0: 0.3,
+        rho: 0.003,
+        tau: 0.2,
+        decentralisation_param: 0,
+        extra_entropy: null,
+        protocol_major_ver: 9,
```
Comment (slowbackspace) [inline test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts:23]:
```suggestion
        protocol_major_ver: 10,
```

looks like it's on 10 now

File: test/e2e/tests/get-protocolp-parameters.e2e.test.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { getWebSocketClient } from '../utils/setup-websocket-client.js';
+
+const fixtures = await getFixtures('get-protocol-parameters');
+
+describe('get-protocol-parameters', () => {
+  for (const fixture of fixtures) {
+    test(fixture.testName, async () => {
+      const ws = getWebSocketClient();
+      const { cost_models, ...data } = await ws.sendAndWait('GET_PROTOCOL_PARAMETERS');
+      expect(data).toMatchObject(fixture.result);
```
Comment (slowbackspace) [inline test/e2e/tests/get-protocolp-parameters.e2e.test.ts:12]:
```suggestion
      expect(data).toStrictEqual(fixture.result);
```
I prefer using `toStrictEqual` in this case. The fixture could be missing some data (as it is now missing `cost_models`) which could go unnoticed with a less strict assertion.

File: test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        protocol_minor_ver: 0,
+        min_utxo: '4310',
+        min_pool_cost: '170000000',
+        nonce: '538e6395f3e5c8b415ac054e56ce4bfed41ce39c0fd2890c86d646a4a387b9b9',
+        cost_models_raw: {
+          PlutusV1: expect.arrayContaining([expect.any(Number)]),
+          PlutusV2: expect.arrayContaining([expect.any(Number)]),
+          PlutusV3: expect.arrayContaining([expect.any(Number)]),
+        },
+        price_mem: 0.0577,
```
Comment (slowbackspace) [inline test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts:33]: missing `cost_models`

**Review by slowbackspace — CHANGES_REQUESTED** [review]

File: test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts (hunk 1 above)
Comment (iccicci) [reply test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts]: c9ce170 should solve this

File: test/e2e/tests/get-protocolp-parameters.e2e.test.ts (hunk 2 above)
Comment (iccicci) [reply test/e2e/tests/get-protocolp-parameters.e2e.test.ts]: c9ce170 should solve this

File: test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts (hunk 5 above)
Comment (iccicci) [reply test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts:35]: I wasn't able to find a way to describe the `cost_models` object with combinations of `expect`s: the object is explicitly checked in the test file.

c9ce170 should solve this

Comment (slowbackspace) [reply test/e2e/fixtures/preprod/get-protocol-parameters.e2e.ts:681]: Let's begin with the strictest version and gradually relax the requirements if needed.
I added checking whole `cost_models` https://github.com/blockfrost/blockfrost-websocket-link/pull/272/commits/d9767840cc9bf49f967996f554ea2b9082711d6c

**Review by slowbackspace — APPROVED** [review]

**Pull request merged**